app/src/server/api/routers/datasets.router.ts

```diff
@@ -93,16 +96,35 @@ export const datasetsRouter = createTRPCRouter({
 
       const totalTestingCount = await prisma.datasetEntry.count({
         where: {
-          datasetId: id,
+          datasetId,
           outdated: false,
           split: "TEST",
         },
       });
 
+      const { cost, inputCost, outputCost } = calculateCost(
+        baseModel,
+        0,
+        trainingEntryStats.totalInputTokens - (trainingEntryStats.totalMatchTokens ?? 0),
+        trainingEntryStats.totalOutputTokens,
+      );
+
+      const { cost: costWithoutPruning } = calculateCost(
+        baseModel,
+        0,
+        trainingEntryStats.totalInputTokens,
+        trainingEntryStats.totalOutputTokens,
+      );
+
       return {
+        cost,
+        costWithoutPruning,
```

Comment:
I think the only values we need to calculate and return from this function are those that change when you select/deselect a pruning rule or change a model. I'm open to feedback on this, but I think the other values we're returning from this function (`totalTestingCount` for instance) should be read from another query, as they currently are. That way we don't have to recalculate them every time we change the base model and need to recalculate cost.